plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -978,16 +1043,90 @@ export class ArgoService implements ArgoServiceApi {
     };
 
     let statusText: string = '';
+    try {
+      const response: GetArgoApplicationFetchResponse = (await fetch(
+        `${url}/api/v1/applications/${argoApplicationName}`,
+        options,
+      )) as GetArgoApplicationFetchResponse;
+
+      statusText = response.statusText;
+
+      if (response.status === 200) {
```

Comment:
This doesn't look to do anything since the return outside of this if clause returns the same response

Reply:
Hello, I have been meaning to respond. I know it looks silly, but without this, typescript does not behave correctly. I have attached some screen shots of when I do not separate the response. 

The desired behavior of the type should show that when the api returns 200 we should get the metadata field, and all other response codes should contain a different data structure.

This photo below shows the type behaving as expected, while separating the returns like we are in the code currently.
![Screenshot 2024-04-22 at 12 41 28 PM](https://github.com/RoadieHQ/roadie-backstage-plugins/assets/105800767/1bf3eee1-3075-47e1-977c-3add130bde64)

This photo below shows the type NOT behaving as expected when we remove the separation. 
![Screenshot 2024-04-22 at 12 43 49 PM](https://github.com/RoadieHQ/roadie-backstage-plugins/assets/105800767/281c96e0-1160-4888-b23d-c9d9292bc90b)


Please let me know If you understand that keeping the returns like this helps with the types behaving as the developer would expect.

Kindly.